x-pack/plugins/lens/public/embeddable/expression_wrapper.tsx

```diff
@@ -53,9 +54,28 @@ export interface ExpressionWrapperProps {
 interface VisualizationErrorProps {
   errors: ExpressionWrapperProps['errors'];
   canEdit: boolean;
+  onExpressionError: ExpressionWrapperProps['onExpressionError'];
+  searchSessionId?: string;
 }
 
-export function VisualizationErrorPanel({ errors, canEdit }: VisualizationErrorProps) {
+export function VisualizationErrorPanel({
+  errors,
+  canEdit,
+  onExpressionError,
+  searchSessionId,
+}: VisualizationErrorProps) {
+  // use a combination of sessionid + error messages to decide whether to trigger a rerender
+  const rerenderKey = `${searchSessionId || ''}-${errors
```

Comment:
not sure that I fully understand why we need in include  `searchSessionId` into key. The session id is the same for all panels inside e.g. dashboard.

Reply:
What if we exclude `searchSessionId` from the key, and do: 

```
if (rerenderKey && keyRef.current !== rerenderKey) {
  keyRef.current = rerenderKey;
  onExpressionError();
}
```

Reply:
Using the `searchSessionId` as part of the key will give the user the ability to try again to load the panel by clicking the `Refresh` button. Otherwise the panel state will be not be updated.